File: 485-max-consecutive-ones/max-consecutive-ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int len=0;
        int l=0,r=0;
        while(r<nums.length)
        {
            if(nums[r]==1)
            {
                len=Math.max(len,r-l+1);
                r++;
            }
            else
            {
                r++;
                l=r;
            }
        }
        return len;
    }
}

File: 485-max-consecutive-ones/max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
         int tempcount=0;
        int i=0;
        int ans=0;
        while(i<nums.size()){
           
           while(i < nums.size() && nums[i]==1){
                  tempcount++;
                  ans=max(ans,tempcount);
              i++;
           }
           
           tempcount=0;

            while(i < nums.size() && nums[i]==0){
                i++;
            }
        }
        return ans;
    }
};